printer/printer_test.go

```diff
@@ -23,31 +23,55 @@ func TestPrinter(t *testing.T) {
 		if strings.HasSuffix(path, ".in") {
 			inputFile := path
 			expectFile := strings.TrimSuffix(path, ".in") + ".expect"
-
-			inputBB, err := os.ReadFile(inputFile)
-			require.NoError(t, err)
-			expectBB, err := os.ReadFile(expectFile)
-			require.NoError(t, err)
+			expectErrorFile := strings.TrimSuffix(path, ".in") + ".error"
 
 			caseName := filepath.Base(path)
 			caseName = strings.TrimSuffix(caseName, ".in")
 
 			t.Run(caseName, func(t *testing.T) {
-				testPrinter(t, inputBB, expectBB)
+				testPrinter(t, inputFile, expectFile, expectErrorFile)
 			})
 		}
 
 		return nil
 	})
 }
 
-func testPrinter(t *testing.T, input, expect []byte) {
-	f, err := parser.ParseFile(t.Name()+".rvr", input)
+func testPrinter(t *testing.T, inputFile string, expectFile string, expectErrorFile string) {
+	inputBB, err := os.ReadFile(inputFile)
+	require.NoError(t, err)
+
+	f, err := parser.ParseFile(t.Name()+".rvr", inputBB)
+	if expectedError := getExpectedErrorMessage(t, expectErrorFile); expectedError != "" {
+		require.Error(t, err)
+		require.Contains(t, err.Error(), expectedError)
+		return
+	}
+
+	expectBB, err := os.ReadFile(expectFile)
 	require.NoError(t, err)
 
 	var buf bytes.Buffer
 	require.NoError(t, printer.Fprint(&buf, f))
 
-	trimmed := strings.TrimRightFunc(string(expect), unicode.IsSpace)
+	trimmed := strings.TrimRightFunc(string(expectBB), unicode.IsSpace)
 	require.Equal(t, trimmed, buf.String(), "%s", buf.String())
 }
+
+// getExpectedErrorMessage will retrieve an optional expected error message for the test.
+func getExpectedErrorMessage(t *testing.T, errorFile string) string {
```

Comment:
Current implementation of diag.Diagnostics prints out (and %d more diagnostics) so for this testing we are able to validate the first Diagnostic only. This is why this returns string instead of []string like the agent converter tests with an error message per line in the testdata file.